game/hiscore.hh

```diff
@@ -44,17 +48,17 @@ public:
 	  in its valid interval. If one of this conditions is not net a
 	  HiscoreException will be raised.
 	  */
-	void addHiscore(unsigned score, unsigned playerid, unsigned songid, unsigned level, std::string const& track);
+	void addHiscore(unsigned score, PlayerId playerid, SongId songid, unsigned short level, std::string const& track);
 
 	typedef std::vector<HiscoreItem> HiscoreVector;
 
 	/// This queries the database for a sorted vector of highscores. The defaults mean to query everything.
 	/// @param max limits the number of elements returned.
-	HiscoreVector queryHiscore(unsigned max, unsigned playerid, unsigned songid, std::string const& track) const;
-	bool hasHiscore(unsigned songid) const;
+	HiscoreVector queryHiscore(PlayerId playerid, SongId songid, std::string const& track, std::optional<unsigned> max = std::optional<unsigned>()) const;
```

Comment:
Yeah, I had this really dumb moment I began doing this because I couldn't understand why `nullopt`wasn't working... then I realized I was typing `nullopt_t`. Clearly missed this one when I went around fixing it.